Sources/RegexBuilder/DSL.swift

```diff
@@ -186,13 +218,33 @@ public struct ChoiceOf<Output>: _BuiltinRegexComponent {
     self.regex = regex
   }
 
+  /// Creates a regex component that chooses exactly one of the regex components
+  /// provided by the builder closure.
+  ///
+  /// In this example, `regex` successfully matches either a `"CREDIT"` or
+  /// `"DEBIT"` substring:
+  ///
+  ///     let regex = Regex {
+  ///         ChoiceOf {
+  ///             "CREDIT"
+  ///             "DEBIT"
+  ///         }
+  ///     }
+  ///     let match = try regex.prefixMatch(in: "DEBIT    04032020    Payroll $69.73")
+  ///     print(match?.0 as Any)
+  ///     // Prints "DEBIT"
+  ///
+  /// - Parameter builder: A builder closure that declares a list of regex
+  ///   components, each of which can be exclusively matched.
   public init(@AlternationBuilder _ builder: () -> Self) {
     self = builder()
   }
 }
 
 // MARK: - Capture
 
+/// A regex component that saves the matched substring, or a transformed result,
+/// for access in a regular expression match.
```

Comment:
Need an example here that demonstrates capturing and disambiguates from `TryCapture`.